Automattic/wp-calypso
# Remove sessionStorage, mediaModalGalleryInstructionsDismissed -> redux

This PR is continuing moving preferences to redux, as started in https://github.com/Automattic/wp-calypso/pull/5401 . This move is needed to remove `localStorage` dependency

This particular PR also removes `sessionStorage` dependency.

# Functional change

none

## Testing instructions

1. Reset your `mediaModalGalleryInstructionsDismissed `  preference
![zrzut ekranu 2016-06-17 o 13 21 11](https://cloud.githubusercontent.com/assets/3775068/16149087/8ffd6f80-348e-11e6-8a0d-1427be1f381f.png)
2. Load post editor
3. Go to gallery
4. Select item
5. You should be seeing "Help tip"
![zrzut ekranu 2016-06-17 o 12 46 44](https://cloud.githubusercontent.com/assets/3775068/16149124/c8f567de-348e-11e6-9bbc-8290e2fce512.png)
6. *UNTICK* `Don't show again` and click "GOT IT"
7.  *without reloading* Repeat 2-5 and see that tip is gone.
8. reload, repeat 2-5 and see that tip is back
9. tick "Dont show again", click got it
10. Reload, repeat 2-5, see that tip is gone
11. `localStorage.clear()`
12. Reload, repeat 2-5, see that tip is gone

CC @mtias @gwwar @aduth
